### PR TITLE
📖 Build out & fix mirroring example to be more descriptive

### PR DESCRIPTION
**Pull request by stiller-leser**
Hi,

I went and tried to implement the mirroring example. As I am fairly new to the controller-runtime, it took me a while to understand the example. Going through it, there were a few things I was stumbling about:

- `secretMirrorReconciler` and`reconcile.Request` both named `r`
- context missing from `Reconcile` method's parameters
- Position of `secret` creation. It was inside an error block.
- Imports - took a while to find the right imports. Especially `kerrors`, as there are other go packages named this way.
- Additional `}` in `NewSecretMirrorReconciler`

### CODE REVIEW

**Comment by stiller-leser** [comment]
`- Position of `secret` creation. It was inside an error block.`

And it was supposed to. Pushed a fix for that part.

File: designs/move-cluster-specific-code-out-of-manager.md [hunk 1]
```diff
@@ -159,30 +159,49 @@ secret found in `referenceCluster` if none of that name already exists. To keep
...
 	referenceClusterClient, mirrorClusterClient client.Client
 }
 
-func (r *secretMirrorReconciler) Reconcile(r reconcile.Request)(reconcile.Result, error){
+func (r *secretMirrorReconciler) Reconcile(context context.Context, req reconcile.Request)(reconcile.Result, error){
 	s := &corev1.Secret{}
-	if err := r.referenceClusterClient.Get(context.TODO(), r.NamespacedName, s); err != nil {
-		if kerrors.IsNotFound{ return reconcile.Result{}, nil }
-		return reconcile.Result, err
+	if err := r.referenceClusterClient.Get(context.TODO(), req.NamespacedName, s); err != nil {
```
[the remark this answers is not in the document]
Comment (stiller-leser) [reply designs/move-cluster-specific-code-out-of-manager.md]: You are right. I was unware that `context.TODO()` creates a new empty context. I have updated the example.

**Review by vincepri — APPROVED** [review]
/assign @alvaroaleman

**Review by alvaroaleman — COMMENTED** [review]
I'm not really in favor of updating the design to keep up with c-r changes made after the design was implemented and merged. Could you add an example under `examples` instead? That also has the advantage that we actually test the code, so spotting mistakes doesn't rely on the reviewers ability but can be done by the go compiler.

**Comment by stiller-leser** [comment]
Sorry for the late reaction here, I could certainly do that. Unfortunately, with the release of 0.15.0 I have no idea how to do this any longer :(

**Comment by sbueringer** [comment]
Closing this PR for now given no recent activity.

Feel free to reopen the PR if you're still interested to continue this work

/close

**Pull request closed without merging**